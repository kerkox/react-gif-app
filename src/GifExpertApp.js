import React, {useState } from 'react';
import { AddCategory } from './components/AddCategory';
const GifExpertApp = () => {

  const initialState = ['One Punch', 'Samurai X', 'Dragon Ball'];
  const [categories, setCategories] = useState(initialState)

  // const handleAdd = () => {
  //   setCategories(cats => [...cats, 'Jeniffer Rosales']);
  // }

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setCategories}/>
      <hr/>
      <ol>
        { 
          categories.map((category, i) => {
          return <li key={i}>{category}</li>
          })
        }
      </ol>
    </>
  )
}

export default GifExpertApp;