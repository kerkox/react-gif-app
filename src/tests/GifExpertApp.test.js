import { shallow } from "enzyme"
import GifExpertApp from "../GifExpertApp"

describe('Pruebas de <GifExpertApp />', () => {
  test('debe de mostrarse correctamente', () => {
    let wrapper = shallow(<GifExpertApp />)
    expect( wrapper ).toMatchSnapshot()
    
  })

  test('debe de mostrar una lista de categorias', () => {

    const categories = ['One Punch', 'Dragon ball']
    const wrapper = shallow( <GifExpertApp defaultCategories={categories} />)

    expect( wrapper ).toMatchSnapshot()
    expect( wrapper.find('GifGrid').length ).toBe(categories.length)
    
  })
  
  
  
})
