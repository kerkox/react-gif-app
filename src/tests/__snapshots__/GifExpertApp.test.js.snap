// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="0"
    />
    <GifGrid
      category="Dragon ball"
      key="1"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas de <GifExpertApp /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
